incubator/hnc/pkg/reconcilers/hierarchy_config.go

```diff
@@ -423,28 +423,34 @@ func (r *HierarchyConfigReconciler) enqueueAffected(log logr.Logger, reason stri
 }
 
 func (r *HierarchyConfigReconciler) writeInstances(ctx context.Context, log logr.Logger, oldHC, newHC *api.HierarchyConfiguration, oldNS, newNS *corev1.Namespace) (bool, error) {
-	ret := false
-	if updated, err := r.writeHierarchy(ctx, log, oldHC, newHC); err != nil {
+	isDeletingNS := !newNS.DeletionTimestamp.IsZero()
+	updated := false
+	if up, err := r.writeHierarchy(ctx, log, oldHC, newHC, isDeletingNS); err != nil {
 		return false, err
 	} else {
-		ret = ret || updated
+		updated = updated || up
 	}
 
-	if updated, err := r.writeNamespace(ctx, log, oldNS, newNS); err != nil {
+	if up, err := r.writeNamespace(ctx, log, oldNS, newNS); err != nil {
```

Comment:
I am not sure if we still need to update namespace if `isDeletingNS == true`, since the only thing that could change on a namespace is tree labels. I am not sure if dropping the update of a deleting namespace's tree labels could cause any issue for hierarchical resource quota since it may use tree labels.

Reply:
I'm a little nervous about stopping this update. I agree that the labels could disappear from the namespace shortly before it gets deleted, but this should be transient and very rare.

Maybe we can file an issue to revisit?

Reply:
Ping - @yiqigao217 sgty?

Reply:
SGTM, filed #635 .